tests/integration/dhcp_test.py

```diff
@@ -192,6 +192,126 @@ def test_dhcp_for_bond_with_ip_address_and_slave(dhcp_env,
     assertlib.assert_state(desired_state)
 
 
+def test_ipv4_dhcp_ignore_gateway(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['auto-gateway'] = False
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp gateway got ignored.
+
+
+def test_ipv4_dhcp_ignore_dns(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['auto-dns'] = False
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support DNS query, need to make
+    # sure dhcp DNS entries got ignored.
+
+
+def test_ipv4_dhcp_ignore_routes(dhcp_env):
```

Comment:
It seems that tests that the auto-options actually work are missing.

Reply:
Could you elaborate on this?

Reply:
I mean tests to check if `auto-routes is True` that the routes are actually present in `ip route` output. And the same for the DNS and gateway tests.